python/semantic_kernel/connectors/memory/pinecone/pinecone_memory_store.py

```diff
@@ -143,7 +138,7 @@ async def does_collection_exist(self, collection_name: str) -> bool:
         Returns:
             bool -- True if the collection exists; otherwise, False.
         """
-        return collection_name in self.pinecone.list_indexes()
+        return collection_name in self.pinecone.list_indexes().names()
```

Comment:
any way this could be async, is there a async client available maybe? and then caching this list might also make sense, not sure how other this method get's called, but I see `does_collection_exist` in a lot of places!

Reply:
I did not find an async pinecone client, which is unfortunate due to the asynchronous nature of this codebase.
I added a cache around collection names, good call!